src/models/WorldpayTasks.class.js

```diff
@@ -4,31 +4,38 @@ const formatSimplifiedAccountPathsFor = require('../utils/simplified-account/for
 const paths = require('@root/paths')
 const { ConnectorClient } = require('@services/clients/connector.client')
 const TASK_STATUS = require('@models/constants/task-status')
+const { CREDENTIAL_STATE } = require('@utils/credentials')
 const connectorClient = new ConnectorClient(process.env.CONNECTOR_URL)
 
 class WorldpayTasks {
   /**
    * @param {GatewayAccount} gatewayAccount
    * @param {String} serviceExternalId
+   * @param {Boolean} switchingPsp
    */
-  constructor (gatewayAccount, serviceExternalId) {
+  constructor (gatewayAccount, serviceExternalId, switchingPsp = false) {
```

Comment:
i wonder whether there should be a static `WorldpayTasks.forSwitchingPsp` method instead of a constructor with various options